src/audio/coreaudio/SDL_coreaudio.m

```diff
@@ -769,21 +769,27 @@ static bool PrepareAudioQueue(SDL_AudioDevice *device)
             // L R C LFE Cs Ls Rs
             layout.mChannelLayoutTag = kAudioChannelLayoutTag_WAVE_6_1;
         } else {
-            // FIXME: We need to manually swizzle channels into a supported layout
             // L R C LFE Ls Rs Cs
-            //layout.mChannelLayoutTag = kAudioChannelLayoutTag_MPEG_6_1_A;
-            return SDL_SetError("Unsupported audio channels");
+            layout.mChannelLayoutTag = kAudioChannelLayoutTag_MPEG_6_1_A;
+
+            int swizzle_map[7] = {
+                0, 1, 2, 3, 6, 4, 5
+            };
+            device->chmap = SDL_ChannelMapDup(swizzle_map, SDL_arraysize(swizzle_map));
```

Comment:
These should check if SDL_ChannelMapDup returned NULL and return false (SDL_SetError is already called, it's out of memory).

Returning false will cause SDL to clean up all the half-initialized stuff, so you don't have to add SDL_free() calls to the rest of the function (or free the chmap when closing the device).